chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -753,22 +757,47 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             // Check tokens
             tx.outputs()
                 .iter()
-                .filter_map(|output| match output {
+                .try_for_each(|output| match output {
                     TxOutput::Transfer(v, _)
                     | TxOutput::LockThenTransfer(v, _, _)
-                    | TxOutput::Burn(v) => v.token_data(),
+                    | TxOutput::Burn(v) => {
+                        if let Some(token_data) = v.token_data() {
+                            check_tokens_data(
+                                self.chain_config,
+                                token_data,
+                                tx.transaction(),
+                                block.get_id(),
+                            )?;
+                        }
+                        Ok(())
+                    }
+                    TxOutput::Tokens(token_output) => match token_output {
+                        TokenOutput::IssueFungibleToken(issuance) => check_tokens_issuance(
+                            self.chain_config,
+                            issuance.as_ref(),
```

Comment:
I think &issuance works fine instead of explicitly as_ref()... it'll simplify the formatting.